packages/contracts-bedrock/src/L2/OptimismSuperchainERC20.sol

```diff
@@ -0,0 +1,166 @@
+
+/// @notice Thrown when attempting to relay a message and the cross domain message sender is not this
+/// OptimismSuperchainERC20.
+error MessageSenderNotThisSuperchainERC20();
+
+/// @notice Thrown when attempting to mint or burn tokens and the function caller is not the StandardBridge.
+error CallerNotBridge();
+
+/// @notice Thrown when attempting to mint or burn tokens and the account is the zero address.
+error ZeroAddress();
+
+/// @custom:proxied
+/// @title OptimismSuperchainERC20
+/// @notice OptimismSuperchainERC20 is a standard extension of the base ERC20 token contract that unifies ERC20 token
+///         bridging to make it fungible across the Superchain. This construction builds on top of the
+///         L2ToL2CrossDomainMessenger for both replay protection and domain binding.
+contract OptimismSuperchainERC20 is ERC20, ISemver {
+    /// @notice Address of the corresponding version of this token on the remote chain.
+    address public immutable REMOTE_TOKEN;
+
+    /// @notice Decimals of the token
+    uint8 private immutable DECIMALS;
+
+    /// @notice Name of the token
+    string private _name;
+
+    /// @notice Symbol of the token
+    string private _symbol;
+
+    /// @notice Emitted whenever tokens are minted for an account.
+    /// @param account Address of the account tokens are being minted for.
+    /// @param amount  Amount of tokens minted.
+    event Mint(address indexed account, uint256 amount);
+
+    /// @notice Emitted whenever tokens are burned from an account.
+    /// @param account Address of the account tokens are being burned from.
+    /// @param amount  Amount of tokens burned.
+    event Burn(address indexed account, uint256 amount);
+
+    /// @notice Emitted whenever tokens are sent to another chain.
+    /// @param from    Address of the sender.
+    /// @param to      Address of the recipient.
+    /// @param amount  Amount of tokens sent.
+    /// @param chainId Chain ID of the destination chain.
+    event SentERC20(address indexed from, address indexed to, uint256 amount, uint256 chainId);
+
+    /// @notice Emitted whenever tokens are successfully relayed on this chain.
+    /// @param to     Address of the recipient.
+    /// @param amount Amount of tokens relayed.
+    event RelayedERC20(address indexed to, uint256 amount);
+
+    /// @notice A modifier that only allows the bridge to call
+    modifier onlyBridge() {
+        if (msg.sender != Predeploys.L2_STANDARD_BRIDGE) revert CallerNotBridge();
+        _;
+    }
+
+    /// @notice Semantic version.
+    /// @custom:semver 1.0.0-beta.1
+    string public constant version = "1.0.0-beta.1";
+
+    /// @param _remoteToken     Address of the corresponding remote token.
+    /// @param _tokenName       ERC20 name.
+    /// @param _tokenSymbol     ERC20 symbol.
+    /// @param _decimals        ERC20 decimals.
+    constructor(address _remoteToken, string memory _tokenName, string memory _tokenSymbol, uint8 _decimals) {
```

Comment:
is using `_tokenName` over `__name` something from their style guide or did you find it more tidy?

Reply:
yeah, i found it more tidy, their styles say:

> Function parameters should be prefixed with an underscore.